File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/activities/MainActivity.java
package se.gustavkarlsson.aurora_notifier.android.activities;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;

import se.gustavkarlsson.aurora_notifier.android.R;

public class MainActivity extends AppCompatActivity {

	private static final String TAG = MainActivity.class.getSimpleName();

	@Override
    protected void onCreate(Bundle savedInstanceState) {
		Log.v(TAG, "onCreate");
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
		Log.v(TAG, "onCreateOptionsMenu");
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
		Log.v(TAG, "onOptionsItemSelected");
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/gui/activities/main/MainActivity.kt
package se.gustavkarlsson.skylight.android.gui.activities.main

import android.arch.lifecycle.LifecycleObserver
import android.os.Bundle
import se.gustavkarlsson.skylight.android.R
import se.gustavkarlsson.skylight.android.extensions.findNavController
import se.gustavkarlsson.skylight.android.extensions.setupActionBarWithNavController
import se.gustavkarlsson.skylight.android.gui.activities.AuroraRequirementsCheckingActivity


class MainActivity : AuroraRequirementsCheckingActivity(), LifecycleObserver {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)
		setupActionBarWithNavController(findNavController())
		ensureRequirementsMet()
	}

	override fun onSupportNavigateUp(): Boolean =
		findNavController().navigateUp()

	private fun findNavController() = findNavController(R.id.mainNavHost)

	override fun onRequirementsMet() = Unit
}
